From 55b93f1ff37613c95ed37e0ed305f4ab4c5a05ef Mon Sep 17 00:00:00 2001
From: Dinesh Kumar Mirche <dmirche@habana.ai>
Date: Fri, 19 Aug 2022 10:56:10 +0300
Subject: [PATCH] [SW-95670] gaudi2: increase hbm bar size to 128gb

BAR size for HBM is set to 128GB instead of 64GB.
This is required for host to access of all available
space, so that host won't need to shift small BAR window
for every access which is outside the window.
Therefore, this method is more efficient and more secured.

Change-Id: I3e41579565f3e61e730a59f13df7c014663b0596
Signed-off-by: Dinesh Kumar Mirche <dmirche@habana.ai>
---

diff --git a/arch/gaudi2/common/asic.c b/arch/gaudi2/common/asic.c
index 1bd6eda..666a064 100644
--- a/arch/gaudi2/common/asic.c
+++ b/arch/gaudi2/common/asic.c
@@ -30,10 +30,10 @@
 
 __weak void preboot_update_bar_sizes(void)
 {
-	/* Set HBM BAR size to 64GB */
+	/* Set HBM BAR size to 128GB */
 	regmap_writel(HBM_BAR_PF_NUM << PCIE_AUX_DBI_PF_NUM_SHIFT, mmPCIE_AUX_DBI);
 	regmap_writel(0xFFFFFFFF, mmPCIE_DBI_BAR4_REG);
-	regmap_writel(0x0000000F, mmPCIE_DBI_BAR5_REG);
+	regmap_writel(0x0000001F, mmPCIE_DBI_BAR5_REG);
 	regmap_writel(0x00000000, mmPCIE_AUX_DBI);
 }
 
